### PR TITLE
Add prettier code formatting

### PR DESCRIPTION
**Pull request by jason-fox**
Adds [prettier](https://prettier.io) and  [husky](https://github.com/typicode/husky). All JavaScript files are autoformated on commit using the Git Hooks hook. 

running `npm run prettier` will format all files with standardized whitespace.

### CODE REVIEW

File: bin/iotaJsonTester.js [hunk 1]
```diff
@@ -135,33 +136,36 @@ function exitClient() {
 }
 
 var commands = {
-    'config': {
+    config: {
         parameters: ['host', 'port', 'apiKey', 'deviceId'],
-        description: '\tConfigure the client to emulate the selected device, connecting to the given host.',
+        description:
+            '\tConfigure the client to emulate the selected device, connecting to the given host.',
```
Comment (fgalan) [inline bin/iotaJsonTester.js]: I have a kind of deja vu looking to this... :)

We were discussing about line length in https://github.com/telefonicaid/perseo-fe/pull/314#issuecomment-459038580 and we ended with `"printWidth": 120` (https://github.com/telefonicaid/perseo-fe/blob/master/.prettierrc.json#L6). I think this setting should be common to this an other similar opened PRs.

Comment (fgalan) [reply bin/iotaJsonTester.js]: (I didn't realize in the IOTA Sigfox onboarding PR... the 100 at https://github.com/telefonicaid/sigfox-iotagent/blob/master/.prettierrc.json#L6 probably should be 120 but it's fine given that the PR there is already merged).

Comment (jason-fox) [reply bin/iotaJsonTester.js]: Fixed f90c273

File: package-lock.json [hunk 2]
```diff
@@ -0,0 +1,4820 @@
+{
```
Comment (fgalan) [inline package-lock.json]: IOTA repositories intentionally doesn't include package-lock.json. We can open a discussion thread about the convenience of having pacakges-lock.json included in the repository or not :) but by the moment I'll suggest to leave it out this PR, please.

File: test/unit/startup-test.js [hunk 3]
```diff
@@ -165,19 +169,21 @@ describe('Startup tests', function() {
...
-                    host: '127.0.0.1',
-                    keepalive: 0,
-                    key: 'keycontent',
-                    password: null,
-                    port: '8883',
-                    protocol: 'mqtts',
-                    rejectUnauthorized: true,
-                    username: null
-                }).should.equal(true);
+                mqtt.connect
```
Comment (fgalan) [inline test/unit/startup-test.js]: Not sure if I like this... too "sparse", don't you think?

On the contrary

```
mqtt.connect.calledOnceWithExactly({
```

looks nice.

File: lib/bindings/MQTTBinding.js [hunk 4]
```diff
@@ -98,9 +95,18 @@ function sendConfigurationToDevice(apiKey, deviceId, results, callback) {
     config.getLogger().debug(context, 'Sending requested configuration to the device:\n %j', configurations);
 
     mqttClient.publish(
-        '/' + apiKey + '/' + deviceId + '/' + constants.CONFIGURATION_SUFIX + '/' +
-        constants.CONFIGURATION_VALUES_SUFIX,
-        JSON.stringify(configurations), options, callback);
+        '/' +
```
Comment (fgalan) [inline lib/bindings/MQTTBinding.js]: Topic/URL composition is better in just one line. Could this change be avoided?

File: lib/errors.js [hunk 5]
```diff
@@ -40,12 +40,21 @@ module.exports = {
...
+        this.message =
+            /*jshint quotmark: double */
+            "Some of the mandatory params weren't found in the request: " +
+            /*jshint quotmark: single */
+            JSON.stringify(paramList);
         this.code = 400;
     },
     UnsupportedType: function(expectedType) {
         this.name = 'UNSUPPORTED_TYPE';
-        this.message = 'The request content didn\'t have the expected type [' + expectedType + ' ]';
```
Comment (fgalan) [inline lib/errors.js]: This should be morer compact. Maybe using [%s] it could be done.

File: bin/iotaJsonTester.js [hunk 6]
```diff
@@ -135,33 +136,34 @@ function exitClient() {
...
-    'singleMeasure': {
+    singleMeasure: {
         parameters: ['attribute', 'value'],
         description: '\tSend the given value for the selected attribute to the MQTT broker.',
         handler: checkConnection(singleMeasure)
     },
-    'multipleMeasure': {
+    multipleMeasure: {
         parameters: ['attributes'],
-        description: '\tSend a collection of attributes to the MQTT broker, using JSON format. The "attributes"\n' +
```
Comment (fgalan) [inline bin/iotaJsonTester.js]: This line is 117 chars long. I wonder why prettier has changed it if printWidth is 120...

File: package-lock.json (hunk 2 above)
Comment (jason-fox) [reply package-lock.json]: Package lock removed. Fixed 6f5a131

File: bin/iotaJsonTester.js (hunk 6 above)
Comment (jason-fox) [reply bin/iotaJsonTester.js]: Presumably prettier picks up on the `\n` and breaks at that point

File: lib/errors.js (hunk 5 above)
Comment (jason-fox) [reply lib/errors.js]: Fixed 7657825

File: lib/bindings/MQTTBinding.js (hunk 4 above)
Comment (jason-fox) [reply lib/bindings/MQTTBinding.js]: Fixed 7657825 - Note the line is over 120 chars unless split once anyway.

File: test/unit/startup-test.js (hunk 3 above)
Comment (jason-fox) [reply test/unit/startup-test.js]: Fixed 7657825

File: bin/iotaJsonTester.js (hunk 6 above)
Comment (fgalan) [reply bin/iotaJsonTester.js]: A bit weird... but ok.

NTC

**Comment by dcalvoalonso** [comment]
LGTM! Thanks @jason-fox !

**Pull request merged**